src/frontend/context/cartContext.js

```diff
@@ -0,0 +1,110 @@
+import { createContext, useEffect, useState, useRef } from 'react';
+import axios from 'axios';
+export const CartContext = createContext()
+
+const CartContextProvider = (props) => {
+    const [cart, setCart] = useState([])
+
+    const deleteItem = (id) => {
+        setCart(cart.filter(cart => cart._id !== id))
+    }
+
+    let token = localStorage.getItem("userToken")
+    console.log(token);
+
+    const increaseItem = (id) => {
+        console.log(id);
+        axios.post("/api/user/cart/" + id.toString(), {
+            action: {
+                type: "increment"
+            }
+        }, {
+            headers: { authorization: token }
+        }).then(res => {
+            console.log("incr", res);
+            setCart(res.data.cart)
+        }).catch((error) => {
+            console.log(error.response.data)
+        })
+    }
+
+
+    const addItem = (product) => {
+
+        // if (token != null) {
+        // setisLogged(true)
+        console.log('add', product._id);
+        const found = cart.some(item => item._id === product._id);
+        console.log(found, cart);
+        if (!found) {
+            let token = localStorage.getItem("userToken")
+            console.log(token);
+            axios.post("/api/user/cart", { product }, {
+                headers: { authorization: token }
+            }).then(res => {
+                console.log(res);
+                setCart(res.data.cart)
+            }).catch((error) => {
+                // here you will have access to error.response
+                console.log(error.response.data)
+            })
+        } else {
+            increaseItem(product._id)
+        }
+        // }
+
+    };
+
+
+    useEffect(() => {
+        axios.get("/api/user/cart", {
+            headers: { authorization: token }
+        }).then(res => {
+            console.log(res.data);
+            setCart(res.data.cart)
+        }).catch((error) => {
+            // here you will have access to error.response
+            console.log(error.response.data)
+        })
+    }, [])
+
+    const decreaseQuantity = (product) => {
```

Comment:
suggestion:- you can keep this function in utils folder for better readability , otherwise it is also fine

Reply:
thank you for the feedback, I'll make the changes.